### PR TITLE
Building a Deep Learning Web App using Python

### LINKED ISSUE #4034: [Machine Learning] Building a Deep Learning Selfie Segmentation Web App using Gradio
#### Proposed title of article
Your title should be descriptive of the article/tutorial. Be Specific. 

[Machine Learning] Building a Deep Learning Selfie Segmentation Web App using Gradio

Use keyword research to improve your article's SEO.

#### Proposed article introduction

In this tutorial, we are going to build a selfie segmentation model using Mediapipe, Gradio and Python. The selfie segmentation model allows us to quickly and easily strip out the background from photos using a pre-built machine learning model.

We’re going to take it one step further and integrate it to a Gradio app!

#### Key takeaways 
What are the 3-5 most important things the reader should understand or be able to do after reading this article?
Use this area to get your ideas down on the bulk of your article or tutorial.

By the end of this tutorial, a reader will be able to:
- Install Mediapipe and Gradio
- Apply selfie segmentation using OpenCV
- Integrate the selfie model into the Gradio app

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

- [MediaPipe](https://mediapipe.dev/)
- [Gradio](https://www.gradio.app/)

### PR DESCRIPTION
**Pull request by lcheptoo**
This closes #4034

### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #issue-number`

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

File: content/articles/building-a-deep-learning-selfie-segmentation-web-app-using-gradio/index.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+grid = gridspec.GridSpec(1,2)
+
+# Setup axes
+ax0 = plt.subplot(grid[0])
+ax1 = plt.subplot(grid[1])
+
+ax0.imshow(frame)
+ax1.imshow(res.segmentation_mask)
+plt.show()
+
```
Comment (Bayler) [inline content/articles/building-a-deep-learning-selfie-segmentation-web-app-using-gradio/index.md:123]: Same case with this section.

**Comment by Bayler** [comment]
Hi @lcheptoo Please explain the last three sections in detail. Writing code is not enough. A reader needs to know what the code is doing. Ping me when done.

**Comment by hectorkambow** [comment]
@lcheptoo 
Good afternoon and thank you for submitting your topic to the EngEd program.  It seems this PR is still quite a bit aways from being completed. 

As a rule of thumb - please only submit articles (pull requests) with a complete and fully polished article - ready to be published. 
All revisions and edits should be completed on your own forked repo (as to not build up a backlog in the queue). 

We will be closing this and please re-submit when it is ready to be published as a new PR (and you can add the duplicate label) and ping the reviewer and myself. 
cc @Bayler

**Comment by WanjaMIKE** [comment]
Merged #4141 into master

**Pull request merged**